web/src/pages/apd/new/ApdNew.js

```diff
@@ -0,0 +1,210 @@
+  const [name, setName] = useState(false);
+  const [setIsLoading] = useState(false);
+
+  const createNew = () => {
+    setIsLoading(true);
+    create();
+  };
+
+  const { control, trigger } = useForm({
+    defaultValues: {
+      apdType
+    },
+    mode: 'onBlur',
+    reValidateMode: 'onBlur'
+  });
+
+  const thisFFY = (() => {
+    const year = new Date().getFullYear();
+
+    // Federal fiscal year starts October 1,
+    // but Javascript months start with 0 for
+    // some reason, so October is month 9.
+    if (new Date().getMonth() > 8) {
+      return year + 1;
+    }
+    return year;
+  })();
+
+  const yearOptions = [thisFFY, thisFFY + 1, thisFFY + 2].map(y => `${y}`);
+  const years = yearOptions.slice(0, 2);
+
+  const yearChoices = yearOptions.map((year, index) => ({
+    defaultChecked: years.includes(year),
+    name: `years[${index}].year`,
+    label: year,
+    value: year
+  }));
+
+  const initialState = {
+    apdType,
+    name,
+    years: years
+  };
+
+  const setYears = value => {
+    try {
+      let index = years.indexOf(value);
+      if (index < 0) {
+        let newYears = years.push(value);
+        initialState.years = newYears;
+      } else {
+        years.splice(index);
+      }
+    } catch {
+      err => console.log(err);
+    }
+  };
+
+  const validate = createApdSchema.validate(initialState);
+
+  return (
+    <Fragment>
+      <div className="ds-u-margin--7 ds-u-padding--7 ds-u-padding-right--12">
+        <h1 className="ds-h2">Create a New Advanced Planning Document (APD)</h1>
+        <div className="ds-u-padding-bottom--1 ds-u-border-bottom--2">
+          Complete all the fields below to create your APD.
+        </div>
+        <h2 className="ds-h3">What type of APD are you creating?</h2>
+        <Alert
+          variation="warn"
+          className="ds-u-margin-y--3 ds-u-margin-right--7"
+        >
+          <p className="ds-c-alert__text">
+            This selection cannot be changed after creating a new APD.
+          </p>
+        </Alert>
+        <Controller
+          name="apdType"
+          control={control}
+          render={({ field: { onChange, value, ...props } }) => (
+            <ChoiceList
+              {...props}
+              type="radio"
+              choices={[
+                {
+                  label: 'HITECH IAPD',
+                  value: 'hitech',
+                  checked: value === 'hitech'
+                },
+                {
+                  label: 'MMIS IAPD',
+                  value: 'mmis',
+                  checked: value === 'mmis'
+                }
+              ]}
+              onChange={e => {
+                onChange(e);
+                setApdType(e.target.value);
+              }}
+            />
+          )}
+        />
+        {(apdType === 'mmis' || apdType === 'hitech') && (
+          <div>
+            <Controller
+              name="name"
+              control={control}
+              value=""
```

Comment:
don't need value="", you could get value out of the field below and explicitly set it to textfield

Reply:
If I remove `value=""` it throw the error:

```
A component is changing an uncontrolled input to be controlled. This is likely caused by the value changing from undefined to a defined value, which should not happen. Decide between using a controlled or uncontrolled input element for the lifetime of the component. 
```

Is that okay?